src/ReductionMsg.chpl

```diff
@@ -972,15 +972,18 @@ module ReductionMsg
        agg.copy(IVi, firstIV[idx]);
       }
       var sortedKV: [kD] (int, int);
-      forall (kvi, idx) in zip(sortedKV, IV) with (var agg = newSrcAggregator(int)) {
-        agg.copy(kvi[1], keys[idx]); 
-        agg.copy(kvi[2], values[idx]);
+      forall ((kvi0,kvi1), idx) in zip(sortedKV, IV) with (var agg = newSrcAggregator(int)) {
+        agg.copy(kvi0, keys[idx]);
+        agg.copy(kvi1, values[idx]);
       }
       if v {writeln("sort time = ", Time.getCurrentTime() - t1); try! stdout.flush();}
       if v {writeln("Finding unique (key, value) pairs..."); try! stdout.flush();}
       var truth: [kD] bool;
       // true where new (k, v) pair appears
-      [(t, kv, i) in zip(truth, sortedKV, kD)] if i > kD.low { t = (sortedKV[i-1][2] != kv[2]); }
+      [(t, (_,val), i) in zip(truth, sortedKV, kD)] if i > kD.low {
```

Comment:
here is the other change which might have caused the error

Reply:
Aha, with another look, it looks like I made a transcription error here (note that sortedVal isn't used).  Testing a fix now.

Reply:
Got a fix for 1.20, but still don't have it passing with 1.22.  Timing out for tonight, will try again tomorrow.

Reply:
Latest info: It seems to be a bug of omission rather than a problem with any of the changes that were made.  This hypothesis is based on stepping back to the first commit on this branch in which I merely changed from 1-based to 0-based indexing for Chapel master, and still see the same failures.  So now to find the tuple access I overlooked...

Reply:
OK, I think I've got it now.  I had overlooked one tuple indexing case unfortunately.